### PR TITLE
CE-2981 - Enable shallow clone across all projects

### PR DESCRIPTION
**Pull request by Evildethow**

### CODE REVIEW

File: managed-scripts/groovy/enableShallowCloneOnAllAbstractProjects.groovy [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+import hudson.plugins.git.extensions.impl.*
+
+Jenkins.instance.getAllItems(AbstractProject.class)
+.findAll { job -> job.isBuildable()}
+.findAll { job -> job.scm != null && job.scm instanceof GitSCM}
+.each { project -> 
+  scm = project.scm
+  
+  scm.extensions
+  .findAll {it instanceof CloneOption}
```
Comment (Dohbedoh) [inline managed-scripts/groovy/enableShallowCloneOnAllAbstractProjects.groovy]: So this actually modifies *existing `CloneOption` extension* already added to the GitSCM. What if there is no `CloneOption` extension ?

```
def cloneOption = scm.extensions.find {it instanceof CloneOption}
if (!cloneOption) {
    cloneOption = new CloneOption(true, false, "", 10)
} else {
    cloneOption = new CloneOption(true, it.noTags, it.reference, it.timeout)
}
scm.extensions.remove(it)
scm.extensions.add(cloneOption) 
```

File: managed-scripts/groovy/enableShallowCloneOnAllWorkflowProjects.groovy [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+
+Jenkins.instance.getAllItems(WorkflowJob.class)
+.each { project ->   
+  scm = project.definition.scm 
+  
+  scm.extensions
+  .findAll {it instanceof CloneOption}
+  .findAll {! it.shallow}
+  .each {
+    boolean shallowClone = true
```
Comment (Dohbedoh) [inline managed-scripts/groovy/enableShallowCloneOnAllWorkflowProjects.groovy]: Same here

**Comment by Evildethow** [comment]
@aheritier can you find someone w/ merge permission?

**Comment by aheritier** [comment]
@Evildethow in general it was @imod but I'm not seeing him active anymore

https://jenkins.io/doc/developer/publishing/source-code-hosting/ is saying that @abayer @mrdfuse and @garethbowles could do it too

**Comment by imod** [comment]
@aheritier @Evildethow I'll try to get this done in the next two/three days...

**Comment by aheritier** [comment]
thanks a lot @imod !!

**Pull request merged**